include/boost/math/tools/config.hpp

```diff
@@ -49,6 +49,17 @@
 #define BOOST_STRINGIZE(X) BOOST_DO_STRINGIZE(X)
 #define BOOST_DO_STRINGIZE(X) #X
 
+// Detect thread support via STL implementation
```

Comment:
Thanks Matt, however I suspect this is going to cause macro-redefinition conflicts unless guarded by a check that we're in standalone mode.

BTW BOOST_DISABLE_THREADS, while set inside Boost.Config is intended as a macro the user can set on the command line to force thread support off (for example if they just don't need it, even though it's supported, see https://www.boost.org/doc/libs/1_76_0/libs/config/doc/html/index.html#boost_config.configuring_boost_for_your_platform.user_settable_options), so there should be a check to see it's set already and configure appropriately.

A small point of pedantry: most std libraries will supply `<mutex>` etc even though they may be empty or non-functional in the current compilation mode.  Which is to say further down the road we may need to pull in some of the logic from Boost.Config here for popular platforms at least.

I might also be inclined to set BOOST_NO_CXX11_HDR_ATOMIC, BOOST_NO_CXX11_HDR_FUTURE, BOOST_NO_CXX11_HDR_MUTEX, BOOST_NO_CXX11_HDR_THREAD and BOOST_NO_CXX11_THREAD_LOCAL when BOOST_DISABLE_THREADS is set so that in code we can support partial conformance if needs be.  So I guess it would look something like:

```
#ifdef BOOST_MATH_STANDALONE
#ifdef BOOST_DISABLE_THREADS
// no threads!
#elif defined(__has_include)
#  if !__has_include(<thread>) || !__has_include(<mutex>) || !__has_include(<future>) || !__has_include(<atomic>)
#     define BOOST_DISABLE_THREADS
#  else
#     define BOOST_HAS_THREADS
#  endif 
#else
#  define BOOST_HAS_THREADS // The default assumption is that the machine has threads
#endif // Thread Support
//
// Compiler specific config may go here later....
//
// When threads are disabled, set the individual Boost.Config thread related macros:
//
#ifdef BOOST_DISABLE_THREADS
#  define BOOST_NO_CXX11_HDR_ATOMIC
#  define BOOST_NO_CXX11_HDR_FUTURE
#  define BOOST_NO_CXX11_HDR_MUTEX
#  define BOOST_NO_CXX11_HDR_THREAD
#  define BOOST_NO_CXX11_THREAD_LOCAL
#endif // BOOST_DISABLE_THREADS
#endif // BOOST_MATH_STANDALONE
```